### PR TITLE
Updated `HuggingFaceHubPromptDriver` to use `transformers`'s `apply_chat_template`.

### PR DESCRIPTION
**Pull request by collindutter**
----
📚 Documentation preview 📚: https://griptape--833.org.readthedocs.build//833/

### CODE REVIEW

File: docs/griptape-framework/drivers/prompt-drivers.md [hunk 1]
```diff
@@ -237,52 +237,27 @@ agent.run('Briefly explain how a computer works to a young child.')
...
-    prompt_lines.append("Girafatron:")
-
-    return "\n".join(prompt_lines)
-
-
 agent = Agent(
     config=StructureConfig(
         prompt_driver=HuggingFaceHubPromptDriver(
-            model="tiiuae/falcon-7b-instruct",
+            model="HuggingFaceH4/zephyr-7b-beta",
```
Comment (dylanholmes) [inline docs/griptape-framework/drivers/prompt-drivers.md:259]: Just curious, any particular reason for choosing `HuggingFaceH4/zephyr-7b-beta`?

File: griptape/drivers/prompt/huggingface_hub_prompt_driver.py [hunk 2]
```diff
@@ -61,11 +61,18 @@ def try_run(self, prompt_stack: PromptStack) -> TextArtifact:
...
 
         for token in response:
             yield TextArtifact(value=token)
+
+    def __to_prompt(self, prompt_stack: PromptStack) -> str:
+        tokens = self.tokenizer.tokenizer.apply_chat_template(
+            [{"role": i.role, "content": i.content} for i in prompt_stack.inputs], add_generation_prompt=True
+        )
+
+        return self.tokenizer.tokenizer.decode(tokens)
```
Comment (dylanholmes) [inline griptape/drivers/prompt/huggingface_hub_prompt_driver.py:78]: `apply_chat_template` returns a string directly if you pass `tokenize=False`

File: griptape/tokenizers/huggingface_tokenizer.py [hunk 3]
```diff
@@ -17,7 +17,4 @@ class HuggingFaceTokenizer(BaseTokenizer):
     max_output_tokens: int = field(kw_only=True)  # pyright: ignore[reportGeneralTypeIssues]
 
     def count_tokens(self, text: str | list) -> int:
-        if isinstance(text, str):
-            return len(self.tokenizer.encode(text))
-        else:
-            raise ValueError("Text must be a string.")
+        return len(self.tokenizer.encode(text))
```
Comment (dylanholmes) [inline griptape/tokenizers/huggingface_tokenizer.py:20]: Should the type of the `text` parameter be `str | list[str]`?

[Tokenizer.encode()](https://huggingface.co/docs/transformers/en/main_classes/tokenizer#transformers.PreTrainedTokenizer.encode) accepts `str`, `list[str]`, or `list[int]`, but `list[int]` is for a list of tokenized string ids, which I wouldn't expect to be used anywhere in the framework.

**Review by dylanholmes — APPROVED** [review]
Nice work, love see that usage example get greatly simplified.

Just minor comments/suggestions. Hopefully you see them before merging.

File: griptape/drivers/prompt/huggingface_hub_prompt_driver.py (hunk 2 above)
Comment (collindutter) [reply griptape/drivers/prompt/huggingface_hub_prompt_driver.py:78]: Setting up for a future PR that will use the tokens.

File: griptape/tokenizers/huggingface_tokenizer.py (hunk 3 above)
Comment (collindutter) [reply griptape/tokenizers/huggingface_tokenizer.py:20]: This type signature is mainly to appease pyright -- `list[str]` does not work. This signature will be changing in an upcoming PR that simplifies Tokenizers.

File: docs/griptape-framework/drivers/prompt-drivers.md (hunk 1 above)
Comment (collindutter) [reply docs/griptape-framework/drivers/prompt-drivers.md:259]: It was an un-gated model found from [this article](https://huggingface.co/docs/transformers/en/chat_templating#how-do-i-use-chat-templates)

**Pull request merged**